consensus/parlia/parlia.go

```diff
@@ -1697,41 +1696,32 @@ func (p *Parlia) applyTransaction(
 	return nil
 }
 
-// GetJustifiedHeader returns highest justified block's header before the specific block,
-// the attestation within the specific block will be taken into account.
-func (p *Parlia) GetJustifiedHeader(chain consensus.ChainHeaderReader, header *types.Header) *types.Header {
+// GetJustifiedNumberAndHash returns the highest justified block's number and hash on the branch including and before `header`
+func (p *Parlia) GetJustifiedNumberAndHash(chain consensus.ChainHeaderReader, header *types.Header) (uint64, common.Hash, error) {
 	if chain == nil || header == nil {
-		return nil
+		return 0, common.Hash{}, fmt.Errorf("illegal chain or header")
 	}
-
 	snap, err := p.snapshot(chain, header.Number.Uint64(), header.Hash(), nil)
 	if err != nil {
 		log.Error("Unexpected error when getting snapshot",
 			"error", err, "blockNumber", header.Number.Uint64(), "blockHash", header.Hash())
-		return nil
+		return 0, common.Hash{}, err
 	}
 
-	// If there is no vote justified block, then return root or naturally justified block.
 	if snap.Attestation == nil {
-		if header.Number.Uint64() <= naturallyJustifiedDist {
-			return chain.GetHeaderByNumber(0)
+		if p.chainConfig.IsBoneh(header.Number) {
+			log.Warn("once one attestation generated, attestation of snap would not be nil forever basically")
```

Comment:
Would you change the log to the `debug` level?

Reply:
use warn level, it will output much useless logs between bsc boneh update and bc update with vote address, 
                          but can help to provide info for solving problem online after bc update
use debug level, it will not output much useless logs online, 
                          but can't help to provide info for solving problem online after bc update                          

I can switch it to debug level for now, maybe have other opportunity to recover to warn level after bc update